extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -396,6 +407,35 @@ void registerMetricsFromAnnotatedMethods(SmallRyeMetricsRecorder metrics,
         }
     }
 
+    private void findNonOverriddenDefaultMethods(ClassInfo interfaceInfo, List<ClassInfo> subClasses,
+            SmallRyeMetricsRecorder recorder,
+            BeanArchiveIndexBuildItem beanArchiveIndex, JandexMemberInfoAdapter memberInfoAdapter, BeanInfo beanInfo) {
+        // Check for default methods which are NOT overridden by the bean that we are registering metrics for
+        // or any of its superclasses. Register a metric for each of them.
+        interfaceInfo.methods().forEach(method -> {
+            if (!Modifier.isAbstract(method.flags())) { // only take default methods
+                // Here we check that the default method is not overridden by the bean of any of its superclasses,
+                // because in such case it was already picked up while scanning through superclasses and their methods
```

Comment:
Is that true? It seems that you call `findNonOverriddenDefaultMethods` when you traverse the inheritance hierachy, so not all superclasses are processed yet when you call this method. Would it be better to traverse the inheritance hierarchy twice, first to process superclasses, and second to process interfaces?

Reply:
It does not actually matter what exact method the algorithm registered the metric from, the only thing that matters is the name, because here we're only registering the metric, not binding the registration to any interceptor or so.

Also we're talking about the case when the metric annotation is on the concrete bean (and meaning on the class level, not method level), not in the interface, because we don't support that case yet.

Anyway if i take your scenario and move the `@Timed` annotations to the class-level of `A`, then it indeed breaks, because it attempts to register `A.doSomething` twice. Guess we'll need to track "already registered method names" rather than "already visited classes", I'll come up with something...

Reply:
Changed the tracking of "already visited classes" to tracking of "already registered method names". 

It does not matter where exactly the method is taken from (which superclass or interface) if there are multiple options - the metric name is computed from the (concrete) bean name and method name. We just need to see each method at least once.